File: Python/Easy/RomanToInteger.py
class Solution:
    def romanToInt(self, s):
        roman = {
            'I': 1,
            'V': 5,
            'X': 10,
            'L': 50,
            'C': 100,
            'D': 500,
            'M': 1000
        }
        val = 0
        for i in range(0, len(s) - 1):   
            if roman[s[i]] < roman[s[i + 1]]:
                val -= roman[s[i]]
            else:
                val += roman[s[i]]
        return val + roman[s[-1]]

File: Java/Easy/RomanToInteger.java
class Solution {
    public int romanToInt(String s) {
        int integer = 0;
        Map<Character, Integer> romanMapping = new HashMap<Character, Integer>(){{
            put('I', 1);
            put('V', 5);
            put('X', 10);
            put('L', 50);
            put('C', 100);
            put('D', 500);
            put('M', 1000);
        }};
        
        char[] romanWord = s.toCharArray();
        
        for(int i = 0; i < romanWord.length - 1; i++){
            if(romanMapping.get(romanWord[i]) < romanMapping.get(romanWord[i + 1]))
                integer -= romanMapping.get(romanWord[i]);
            else
                integer += romanMapping.get(romanWord[i]);
        }
        
        return integer + romanMapping.get(romanWord[romanWord.length - 1]);
    }
}